Data Processing Pipelines Charter

kbergin opened the pull request:
~**December 8**: Last call for community reviews~
**January 15**: Last call for oversight reviews

kbergin committed:
First pass at a data processing pipelines charter.

kozbo commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+[Team email](mailto:pipelines-team@data.humancellatlas.org): pipelines-team@data.humancellatlas.org
+__HumanCellAtlas#mintteam:__ team discussions, a place to directly communicate to the team, who works on the service and the pipelines, as needed  
+__HumanCellAtlas#secondary-analysis:__ community-focused space for conversations around data processing pipelines that run in the DCP 
+
```
Formatting nit: you will need an extra line after each bullet item to keep them from flowing into each other (on many markdown interpreters).

kbergin replied:
Ah thanks for the tip! Will update.

brianraymor replied:
You can also use an editor like [vscode](https://code.visualstudio.com/) to preview markdown which I find helpful.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+# [HCA Data Processing Pipelines](mailto:pipelines-team@data.humancellatlas.org)
+
+## Description
+The HCA Data Processing Pipelines are cloud-based and portable analysis pipelines that transform the data in the Human Cell Atlas.
```
It might be a good idea to make this purpose more user facing -- "transform" is less important than why we do it, I think. There's lots of "how" in this statement but it could use more "why".

kbergin replied:
I'll try to come up with a way to phrase the 'why' that isn't too general/vague but also doesn't block us from things we may want to do in the future. Thanks

brianraymor replied:
The filename for the charter must be charter.md.

TimothyTickle replied:
Originally the transform part was to point out that they are the point in the system that performs transformative analysis on HCA DCP data. I will update this to be more clear and define it in the definitions. I am not sure I understand the comment on "how" and "why" content of the description. Could you give an example of how you would improve it?

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+The HCA Data Processing Pipelines are cloud-based and portable analysis pipelines that transform the data in the Human Cell Atlas.
+
+## Objectives
+The data processing service aims to develop scientific analysis pipelines for all analyses required on all scientific data types in the Human Cell Atlas Data Coordination Platform (DCP). Pipelines for additional data types, analyses, and improvements to these pipelines will be added iteratively and in collaboration with the HCA Analysis Working Group (AWG) and the community. 
```
> all analyses **required** on all scientific data types in the Human Cell Atlas Data Coordination Platform (DCP)

(emphasis added)

What's required in this context? Who determines it? Obviously there is some nuance here w.r.t. community desires sometimes clashing with our ability to deliver (thinking about normalization, common coordinate alignment, etc).

TimothyTickle replied:
What we work on is aligned to the HCA project, so the HCA project would work with us to define a pipeline roadmap for the HCA project which is what we would work on (which is, in essence, the current process). I can make this more clear.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+The data processing service aims to develop scientific analysis pipelines for all analyses required on all scientific data types in the Human Cell Atlas Data Coordination Platform (DCP). Pipelines for additional data types, analyses, and improvements to these pipelines will be added iteratively and in collaboration with the HCA Analysis Working Group (AWG) and the community. 
+
+## Definitions
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
```
> until **they** have a standardized pipeline

I might rephrase this to "until the community converges to a standardized pipeline". Also: Do we expect one such community pipeline per assay? Many? Can we point to how this process happens?

TimothyTickle replied:
I would prefer not to use that language. The community should give feedback, thoughts, and contribute to pipelines but their convergence does not make a standard HCA pipeline, AWG approval does.
This should be one pipeline per assay because it is to enable the data modality until an official pipeline is installed. (This is all to be learned but this is my first guess).

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+
+## Definitions
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project  
```
Advise on choices or set our priorities? Right now we're leaning on them for both.

TimothyTickle replied:
They have authority over the scientific choices we make, the mint team prioritizes our work (given GG pipeline selection). We, of course, care deeply of what the AWG thinks and use that as a datum in prioritization but authority for prioritization of work specifically falls within the product team.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP
```
Do we want to be explicit here -- that this process should result in a Standardized Pipeline?

TimothyTickle replied:
In light of growing towards supporting community pipelines, onboarding would focus on engineering a pipeline to work in the infrastructure. This would happen to both community and standard pipelines but standard pipelines would need more work beyond onboarding. As we explore community pipelines we may have to update onboarding documents we have to reflect this.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+## In-scope
+* Scoping or review of scoping for new pipelines  
+* Creating all standardized pipelines for scientific data 
+* Reviewing, on-boarding, and potentially creating community pipelines  
```
I think this part is really exciting. Can we unpack what this responsibility implies? When would we create community pipelines, when would we not? What do we hope a community pipeline looks like? Who are we committing to work with? Will there be efforts to generalize this process to make it easier for external users to bring pipelines to the DCP?

kbergin replied:
I agree it would be great to document answers to those questions. 

>When would we create community pipelines, when would we not? 

I think this question is answered in the definition of a `community pipeline` above.

> What do we hope a community pipeline looks like? 

I'm not sure what this means, outside of the statement in the definition above that it is a pipeline that can run in the DCP but is not a `Standard Pipeline`. Could you elaborate?

> Who are we committing to work with? 

Would this be answered by updating the line to:
`* Reviewing, on-boarding, and potentially creating community pipelines for assays prioritized by the HCA DCP Governance Group`

> Will there be efforts to generalize this process to make it easier for external users to bring pipelines to the DCP?

I do think we should do what we can to help with this, especially surrounding documentation and workshops, which I agree should be highlighted explicitly somewhere. @jodihir Do you think that belongs here as well as in the DCP Communications charter, or only here? I'll add that. 

I do not think the mint team will have the bandwidth to do this topic its full justice, however. I do think the Emerald Empire and Cromwell Interactions team could help and do have this on their minds for future prioritization. I don't think it's an immediate priority and I don't know what those efforts would look like, so I don't know that I'd add more detail here than what I mentioned above.

Thoughts? My other idea is that there could be a separate RFC or Process Charter on community pipelines if we want to expand on the idea further in a public forum.

TimothyTickle replied:
I am glad you are excited about this, but it is still not fully understood. I feel we will grow this in the coming year. I am not sure this level of detail is needed in the charter, I would, however, be excited about making other documentation and guidelines that move us towards exercising this concept. If we learn from the community pipeline effort things that would require the charter to be updated, that is a possibility.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+* Reviewing community tools for use in data processing pipelines 
+* Creating tools that support the creation of community pipelines - may include software, documentation or support
+* Updating pipelines ran in the HCA DCP for optimization of cost, time, or scientific benchmarks  
+* Establishing standard benchmarking used to evaluate data processing pipelines  
```
This is a general tool/infrastructure piece (unity), right?

kbergin replied:
This is more around the benchmarking test suites that would be put into Unity, there is a separate charter for Unity itself.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+### Scientific "guardrails"
+
+* Initial versions of standardized pipelines are reviewed by the AWG.  
+* Changes to standardized pipelines that affect the scientific output are reviewed by the AWG, engineering updates that do not affect the scientific output are not reviewed by the AWG.  
```
It could be a good idea to define what we mean by "updates that affect the scientific output".

TimothyTickle replied:
I used the word "change" instead of "affect" that is more restricted in interpretation (hopefully).

ambrosejcarr commented on charters/DataProcessingService/DataProcessingPipelines.md:
```diff
@@ -0,0 +1,70 @@
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay 
+__Standardized Pipeline:__ An official pipeline of the HCA project 
+
+## In-scope
```
Some of the other green box charters call out documentation explicitly -- I think that would be a good idea here, too.

TimothyTickle replied:
Agreed, will add.

ambrosejcarr reviewed (commented):
Sorry for the late review! I've added some comments where I think we might benefit from adding more detail.

kbergin committed:
Update DataProcessingPipelines.md

Adding new lines to keep formatting consistent

brianraymor committed:
Editorial updates to Communication 

Adopt best practices for slack channel format
Add Mailing List and Slack Channels as sections

TimothyTickle committed:
Updates to charter from community review.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+# [HCA Data Processing Pipelines](mailto:pipelines-team@data.humancellatlas.org)
```
There will be a merge conflict with #28 because both charters use: _charters/DataProcessingService/charter.md_ as their subdirectory. This should move to _charters/DataProcessingPipelines/charter.md_?

TimothyTickle replied:
Updated the directory as above.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+[Rex Wang](mailto:chengche@broadinstitute.org)  
+[Saman Ehsan](mailto:sehsan@broadinstitute.org)  
+
+## Communication
```
An earlier [editorial commit](https://github.com/HumanCellAtlas/dcp-community/pull/35/commits/36a8e08251bed0d586acb8887168453a538955f6) was overwritten by the latest update. 

To match the style of other charters, could you replace this section with:

```
### Mailing List
Team email: pipelines-team@data.humancellatlas.org

### Slack Channels
 * [HumanCellAtlas/mintteam](https://humancellatlas.slack.com/messages/mintteam): team discussions, a place to directly communicate to the team, who works on the service and the pipelines, as needed.
 * [HumanCellAtlas/secondary-analysis](https://humancellatlas.slack.com/messages/secondary-analysis): community-focused space for conversations around data processing pipelines that run in the DCP 
```

TimothyTickle replied:
Updated

brianraymor replied:
@TimothyTickle - Almost, but not quite. Could you copy-and-paste the markdown that I included in the [comment](https://github.com/HumanCellAtlas/dcp-community/pull/35#discussion_r235080044)? You only added the additional sub-headings. We're also adding ilnks to slack channels based on a [best practice](https://github.com/HumanCellAtlas/dcp-community/issues/45) introduced by Laura. Your current style for email is also redundant. You end up with two references to the mailto url.

TimothyTickle replied:
copied and pasted.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+## Github repositories
+
+[scTools](https://github.com/HumanCellAtlas/sctools): Tools for single cell data processing  
+[Skylab Analysis](https://github.com/HumanCellAtlas/skylab-analysis): Analysis and benchmarking reports for Skylab pipelines  
```
I'm a new community member reading this charter. What is a _Skylab_ pipeline? It sounds like an internal codename such as _Blue Box_ instead of _Data Store_.

kbergin replied:
It's defined immediately below this and is the name of the repository, so it's really the only name it has. We could swap the order of these definitions if that helps.

brianraymor replied:
Can we simply remove _Skylab_ from the definition of the repo? It doesn't matter what the repo is called as long as the definition does not depend on an internal codename.

TimothyTickle replied:
Updated the language to remove skylab from the definition and focus more on the fact they are standardized HCA data processing pipelines.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+__2018:__ Two data processing pipelines in production (Smart-seq2 and 10xv2 human data)  
+__2019:__ Adding at least three more data processing pipelines, including 10xv2 snRNA-seq and mouse pipelines for all current analysis pipelines Aim to have at least one _in situ_ transcriptomics pipeline in this year. Aim for one of these pipelines to be a community pipeline. Improved automated testing and benchmarking of analysis pipelines  
+__2020:__ Continued addition of analysis pipelines  
```
*2020* is an ambiguous milestone. I would recommend removing it and re-chartering when there are more specifics.

TimothyTickle replied:
Removed.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+## Milestones and Deliverables
+
+__2018:__ Two data processing pipelines in production (Smart-seq2 and 10xv2 human data)  
+__2019:__ Adding at least three more data processing pipelines, including 10xv2 snRNA-seq and mouse pipelines for all current analysis pipelines Aim to have at least one _in situ_ transcriptomics pipeline in this year. Aim for one of these pipelines to be a community pipeline. Improved automated testing and benchmarking of analysis pipelines  
```
_Improved automated testing and benchmarking of analysis pipelines_ - is that not the charter for the Unity Benchmarking service?

Do you have any sense for the rough schedule (Q1, H2 for example) for the 2019 deliverables? One year is an extremely large milestone and seems more aspirational.

*Editorial suggestion*:
**2019:** Add at least three more data processing pipelines, including 10xv2 snRNA-seq and mouse pipelines for all current analysis pipelines. Support at least one _in situ_ transcriptomics pipeline and at least one community pipeline.

kbergin replied:
Unity benchmarking service is a place for the community, including the mint team, to run benchmarking comparisons. The benchmarking comparisons that are run in Unity have to come from somewhere, and the mint team develops them for our pipelines. In addition, we have other testing in mind that could be leveraged by Unity someday in the future but is not currently in their immediate roadmap.

I'd prefer not to have to recharter if our priorities shift throughout the year, re more specific dates.

TimothyTickle replied:
Updated editorial suggestion.

Acknowledging bandwidth for automated testing makes sense because the pipeline generators create the benchmarking pipeline specific to that assay which is then installed and ran in Unity. It is kinda like a very complex checker script. If this feels communicated as a duplication we can focus the language on the benchmarking pipeline development.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+## Objectives
+
+The HCA Data Processing Pipelines team develops all pipelines required by the HCA project for all transformative analyses on scientific data types in the HCA Data Coordination Platform (DCP). Pipelines created to respond to the addition of data types and assays and improvements to pipelines will be added iteratively and in collaboration with the HCA Analysis Working Group (AWG) and the community. ...
```
Editorial suggestion:
_Pipelines are created in response to the introduction of new data types and assays. Iterative improvements to pipelines are made in collaboration with the HCA Analysis Working Group (AWG) and the community._

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+The HCA Data Processing Pipelines team develops all pipelines required by the HCA project for all transformative analyses on scientific data types in the HCA Data Coordination Platform (DCP). Pipelines created to respond to the addition of data types and assays and improvements to pipelines will be added iteratively and in collaboration with the HCA Analysis Working Group (AWG) and the community. ...
+
+## Definitions
```
Would it make sense to define **Pipeline** - stealing some text from @ambrosejcarr:

**Pipeline**: A set of one or more computational stages that transform raw data formats into standard output formats suitable for widespread use.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+## Definitions
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
```
I would move **Standardized Pipeline** before **Community Pipeline** so there's not a forward reference to its definition. 

**Community pipeline** -> **Community Pipeline** for consistency.

 It reads like community pipelines are a _less-good_ stopgap until a different, standardized pipeline is available. My impression was that we would iterate on community pipelines to improve them and then (perhaps) standardize or have a _bakeoff_?

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
```
_Nit_ - _A ~working~ governance group of the Human Cell Atlas P~p~roject_

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
```
*Editorial suggestion*:

_The process of integrating an existing pipeline into the HCA DCP_ -or- _The process of engineering an existing pipeline to integrate into the HCA DCP_

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay.  
```
Might be simplified:

_To better understand the effort required to create a pipeline to process data from a specific assay, available resources and knowledge in a community for that assay are investigated._

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay.  
+__Standardized Pipeline:__ An official pipeline of the HCA project.  
```
What makes it _official_? Would it be more correct to write _A pipeline that has been reviewed and approved by the HCA GG_ - per the comment in **Scientific "guardrails"**?

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+### Pipeline Creation
+
+* Scoping or review of scoping for new pipelines  
+* Suggesting, but not selection of, new standardized pipelines for the HCA DCP  
```
_Suggesting_ to whom? _Recommending to the AWG_ for example?

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+* Scoping or review of scoping for new pipelines  
+* Suggesting, but not selection of, new standardized pipelines for the HCA DCP  
+* Creating all standardized HCA pipelines and documenting them  
```
*Editorial*: _Creating and documenting all standardized HCA pipelines_

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+* Scoping or review of scoping for new pipelines  
+* Suggesting, but not selection of, new standardized pipelines for the HCA DCP  
+* Creating all standardized HCA pipelines and documenting them  
+* Reviewing, on-boarding, and potentially creating community pipelines  
```
Just to clarify: does _potentially creating_ mean that this team might create a community pipeline?

kbergin replied:
Yes, we already have a community pipeline that we made running in the DCP

brianraymor replied:
Sorry, I should have clarified. In my mind, _running_ is different from _creating_.

kbergin replied:
> Yes, we already have a community pipeline that _we made_ running in the DCP

emphasis added

We also created the pipeline.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+* Creating all standardized HCA pipelines and documenting them  
+* Reviewing, on-boarding, and potentially creating community pipelines  
+* Reviewing community tools for use in data processing pipelines  
+* Working with the community to provide feedback on metadata standards for raw data in the HCA DCP to provide the required information to enable processing of the raw data  
```
Perhaps:

 _Working with the community to provide feedback on HCA DCP metadata standards to ensure that required information is available to enable processing of raw data._

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+* Reviewing, on-boarding, and potentially creating community pipelines  
+* Reviewing community tools for use in data processing pipelines  
+* Working with the community to provide feedback on metadata standards for raw data in the HCA DCP to provide the required information to enable processing of the raw data  
+* Evaluating and adopting file formats and associated standards for outputs of pipelines, ensuring the data is accessible with standard tools and interoperable with downstream processes  
```
_outputs of pipelines_, ~ensuring~ to ensure the data is accessible

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+* Working with the community to provide feedback on metadata standards for raw data in the HCA DCP to provide the required information to enable processing of the raw data  
+* Evaluating and adopting file formats and associated standards for outputs of pipelines, ensuring the data is accessible with standard tools and interoperable with downstream processes  
+
+### Pipeline Benchmarking
```
How does this relate to Unity?

kbergin replied:
Answered below.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+### Pipeline Improvement
+
+* Updating pipelines ran in the HCA DCP for optimization of cost, time, or scientific benchmarks  
```
Not sure about the past tense of _ran_ which has an odd ring. How about - _Updating HCA DCP pipelines for optimization ..._

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+* Performing outreach to users on the usability of data in downstream applications  
+* Performing outreach to developers on pipeline development in the HCA DCP  
+* Working with the metadata team and data consumers to ensure the metadata provided about pipelines and the analysis results meet the needs of downstream users  
```
_metadata_ -> _Metadata Schema_ - to match their charter name.

Think it should be **meets** - *the metadata ... meets the needs ...*

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+### Support
+
+* Performing outreach to users on the usability of data in downstream applications  
+* Performing outreach to developers on pipeline development in the HCA DCP  
```
What is the outcome of this outreach to developers?

kbergin replied:
Hopefully better pipelines

brianraymor replied:
That would be great to clarify then - for example - _Performing outreach to developers to collect/identify improvements to pipeline development ..._

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+## Scientific "guardrails"
+
+* Initial versions of standardized pipelines are reviewed by the AWG.  
+* Changes to standardized pipelines that change the scientific output are reviewed by the AWG, engineering updates that do not change the scientific output are not reviewed by the AWG.  
```
Editorial suggestions:

_~Changes~ Improvements to standardized pipelines that change the scientific output must be reviewed by the AWG. Engineering updates that do not change the scientific output do not require review by the AWG._

TimothyTickle replied:
Updated to "Updates to standardized pipelines that change the scientific output must be reviewed by the AWG. Engineering     updates that do not change the scientific output do not require review by the AWG."

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+## Scientific "guardrails"
+
+* Initial versions of standardized pipelines are reviewed by the AWG.  
```
are -> must be

TimothyTickle replied:
updated, thanks!

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+* Initial versions of standardized pipelines are reviewed by the AWG.  
+* Changes to standardized pipelines that change the scientific output are reviewed by the AWG, engineering updates that do not change the scientific output are not reviewed by the AWG.  
+* Selection of standardized pipelines to work on are approved by the HCA DCP Governance Group (this does not apply to community pipelines, community pipelines are onboarded as a service to the scientific community). Prioritization of the work to develop selected pipelines is performed by the Data Processing Service team.  
```
Editorial suggestions:

Selection of standardized pipelines to ~work on~ develop ~are~ must be approved by the HCA DCP Governance Group. (Note, this does not apply to Community Pipelines which are onboarded as a service to the scientific community.) The Data Processing ~Service~ Pipelines team is responsible for setting the development priorities for the selected pipelines.

TimothyTickle replied:
updated.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+
+* Educational training or user assistance on how to perform data analysis with outputs of pipelines  
+* Recommendations or demonstrations of downstream tool use beyond initial data loading  
+* Novel methodological development is not performed by the data processing team unless it is essential to the pipeline, it does not existent in community tools, and it represents minimal effort to implement  
```
Editorial:

data processing team -> Data Processing Pipelines team

it ~does not existent~ is not available in community tools, and ~it represents~ requires minimal effort to implement.

TimothyTickle replied:
updated.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,94 @@
+* Educational training or user assistance on how to perform data analysis with outputs of pipelines  
+* Recommendations or demonstrations of downstream tool use beyond initial data loading  
+* Novel methodological development is not performed by the data processing team unless it is essential to the pipeline, it does not existent in community tools, and it represents minimal effort to implement  
+* Software engineering 3rd party tools  
```
Not sure what this means. You're not planning to contribute software to third party tools?

TimothyTickle replied:
We have had 3rd party tools expect that we would go to extreme efforts to use their tools (do major engineering, implement CI, maintain their repo and such. We preference using 3rd party tools but will not be their dev team. I will update this to give a little more information.

TimothyTickle committed:
Updated directory name

TimothyTickle committed:
Updating communucation section, splitting to more specific resolution

TimothyTickle committed:
Removing ambiguous milestone.

TimothyTickle committed:
Editorial updates and scoep updates.

TimothyTickle committed:
Added communication header

TimothyTickle committed:
Update repos

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+## Definitions
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
```
I think this needs to be more specific. I imagine that there are other parts of the leadership that 'advise and monitor scientific choices' made in HCA. What do the choices concern?

TimothyTickle replied:
Extremely good catch. Would this clarify:
"A group of domain specialists that advise on and monitor scientific choices made when developing the Human Cell Atlas Project Data Coordination Platform."

TimothyTickle replied:
OK that is updated now.

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
```
Here might be an opportunity to call out the Data Processing Pipelines' relationship to the Data Processing Service and link to it. This would make it easier for outsiders to understand the integration of the systems.

TimothyTickle replied:
I like that. Adding.

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay.  
```
That sounds more like estimation than it sounds like scoping work. I would change the definition to something like "[defining the set of features desired](https://en.wikipedia.org/wiki/Project_management_triangle)" if you really mean scoping.

TimothyTickle replied:
This is an odd one, we call this scoping because this investigation gives us the scope of how much work a pipeline will be. You are right it is an investigation. Here we are defining what amount of work will be needed to meet our generic requirements for pipelines.

mweiden replied:
^^ estimation, not investigation.

> Here we are defining what amount of work will be needed to meet our generic requirements for pipelines.

This is, essentially, what I wanted to get at.

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay.  
+__Standardized Pipeline:__ An official pipeline of the HCA project.  
```
I think you need a more precise definition. Isn't a community pipeline that has been accepted, in some ways, "official?"

TimothyTickle replied:
Not official to the HCA project, they are a convenience but not official to the project.

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+__Onboarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
+__Scoping:__ The process of investigating the available resources and knowledge in a community around a specific assay to better understand how much work it would take to make a pipeline to process data from that assay.  
+__Standardized Pipeline:__ An official pipeline of the HCA project.  
+__Transformative Analysis:__ The processing of data that changes data or creates new data. This would not include processes that solely subset or reorganize data.
```
Here too, I think you need a more precise definition. Technically, all DCP components change data or create new data.

TimothyTickle replied:
We have historically used this term/concept for the processing pipelines. Something that generates new data from data as opposed to something that reorganizes or subsets data. Would you have a better suggestion or wording? I am open to suggestions.

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+* Scoping or review of scoping for new pipelines  
+* Suggesting, but not selection of, new standardized pipelines for the HCA DCP  
+* Creating all standardized HCA pipelines and documenting them  
+* Reviewing, on-boarding, and potentially creating community pipelines  
```
nit: [onboarding](https://en.wikipedia.org/wiki/Onboarding)

TimothyTickle replied:
Will update, thanks!

mweiden commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+
+* Investigating and identifying data for pipeline benchmarking  
+* Establishing and operating automated practices around scientific testing of pipelines  
+* Establishing standard benchmarking used to evaluate data processing pipelines  
```
This last bullet seems to overlap with the `In-scope` section of the [Unity Benchmarking Service](https://github.com/HumanCellAtlas/dcp-community/pull/33). Is this about establishing standard benchmarking techniques?

Could this whole section be ported to the Unity charter?

TimothyTickle replied:
Data processing establishes (selected) standard benchmarking datasets (selects the specific datasets), Unity would not select the standard datasets but take them from the community (or data processing team), install them and enforce them in the system.

mweiden reviewed (commented):
Just some comments, mostly intended to make this more accessible to readers new to HCA.

jonahcool commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,98 @@
+## Milestones and Deliverables
+
+__2018:__ Two data processing pipelines in production (Smart-seq2 and 10xv2 human data)  
+__2019:__ Adding at least three more data processing pipelines, including 10xv2 snRNA-seq and mouse pipelines for all current analysis pipelines Aim to have at least one _in situ_ transcriptomics pipeline in this year. Aim for one of these pipelines to be a community pipeline. Improved automated testing and benchmarking of analysis pipelines  
```
```suggestion
__2019:__ Adding at least three more data processing pipelines, including 10xv2 snRNA-seq and mouse pipelines for all current analysis pipelines Aim to have at least one _in situ_ transcriptomics pipeline in this year. Aim for one of these pipelines to be a community pipeline. Improved automated testing and benchmarking of analysis pipelines  
```

jonahcool replied:
Sorry - hit the wrong button. Given that we will be discussing pipeline prioritization with the GG on the December 13th call, do others agree that we should wait to solidify these priorities/deliverables until we have their feedback and are able to integrate it? Perhaps even add a designee of the AWG as a reviewer?

TimothyTickle replied:
I do believe the GG meeting is going to be valuable for feedback that could affect the charter milestones. I am happy to follow whatever process @brianraymor feels is appropriate.

brianraymor replied:
Given that both of you feel that the GG feedback would be valuable, I would recommend treating this as a variation of the normal  _Science Review_ flow described in the charter process:

- Assign the Science Reviewer as a reviewer of the pull request
- Add the "science-review-required" label
- Add a two week last call for the completion of the Science review to the top-level summary comment in the charter pull request.

EXAMPLE
**April 5**: Last call for science review

- When the feedback from the Science review is addressed, replace the "science-review-required" label with "science-review-completed".

@jonahcool - It's also an intriguing proposal to add an AWG member as reviewer.

TimothyTickle replied:
We could do one of two things, confirm our priorities with the GG and make sure the charter is inline after the meeting or ask a GG member for review. Given it is pipeline prioritization it should be the GG reviewing.

TimothyTickle committed:
editorial changes, added definition.

TimothyTickle committed:
Updated AWG definition.

lauraclarke commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,97 @@
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__Data Processing Service:__ A service that executes data processing pipelines on DCP data in the Data Storage Service.
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made when developing the Human Cell Atlas Project Data Coordination Platform.  
```
The AWG advises scientific choices made in pipeline space. At present they don't perform any advisory roles in any other components.

TimothyTickle replied:
As an example, they are also advising matrix service as needed.

lauraclarke commented on charters/DataProcessingPipelines/charter.md:
```diff
@@ -0,0 +1,97 @@
+### Pipeline Creation
+
+* Scoping or review of scoping for new pipelines  
+* Suggesting, but not selection of, new standardized pipelines for the HCA DCP  
```
Changing Suggesting to Proposing feels like a more concrete less vague verb for the process here but this is a nit pick

TimothyTickle replied:
happy to change

TimothyTickle replied:
updated

lauraclarke reviewed (approved):
I have left a couple of nitpick comments about the use of suggest rather than propose and the definition of the AWG. 

I am happy with the document and for it to be merged even if you don't accept these proposals

TimothyTickle committed:
Updated md spacing.

TimothyTickle committed:
Updated suggesting to proposing.

mweiden commented:
Approved in the 2019-01-18 Architecture Call
